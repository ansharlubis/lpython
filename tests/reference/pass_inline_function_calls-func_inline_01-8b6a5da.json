{
    "basename": "pass_inline_function_calls-func_inline_01-8b6a5da",
    "cmd": "lpython --pass=inline_function_calls --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/func_inline_01.py",
    "infile_hash": "6875179ba40c59c22fc55606261bf2c84290b50932f956c1d8152502",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "pass_inline_function_calls-func_inline_01-8b6a5da.stdout",
    "stdout_hash": "115c888b9a7793449dc017ac7298e4376e371280ab31b33b60baf5a5",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}