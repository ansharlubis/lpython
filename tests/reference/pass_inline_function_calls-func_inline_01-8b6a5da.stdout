([1m[35mTranslationUnit[39m[0m 
   ([33mSymbolTable[39m 
      1 
      {
         fib: 
            ([1m[35mFunction[39m[0m 
               ([33mSymbolTable[39m 
                  2 
                  {
                     _lpython_return_variable: 
                        ([1m[35mVariable[39m[0m 
                           2 
                           _lpython_return_variable [1m[32m
                           ReturnVar[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mInteger[39m[0m 8 []) [1m[32m
                           Source[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .false.), 
                     n: 
                        ([1m[35mVariable[39m[0m 
                           2 
                           n [1m[32m
                           In[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mInteger[39m[0m 8 []) [1m[32m
                           Source[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .false.)
                  }) 
               fib [
               ([1m[35mVar[39m[0m 2 [33mn[39m)] [
               ([1m[35mIf[39m[0m 
                  ([1m[35mIntegerCompare[39m[0m 
                     ([1m[35mVar[39m[0m 2 [33mn[39m) [1m[32m
                     Lt[39m[0m 
                     ([1m[35mCast[39m[0m ([1m[35mIntegerConstant[39m[0m [36m2[39m 
                        ([1m[35mInteger[39m[0m 4 [])) [1m[32m
                        IntegerToInteger[39m[0m 
                        ([1m[35mInteger[39m[0m 8 []) ()) 
                     ([1m[35mLogical[39m[0m 4 []) ()) [
                  ([1m[35m=[39m[0m 
                     ([1m[35mVar[39m[0m 2 [33m_lpython_return_variable[39m) 
                     ([1m[35mVar[39m[0m 2 [33mn[39m) ()) 
                  ([1m[35mReturn[39m[0m)] []) 
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 2 [33m_lpython_return_variable[39m) 
                  ([1m[35mIntegerBinOp[39m[0m 
                     ([1m[35mFunctionCall[39m[0m 1 [33mfib[39m () [
                        (
                           ([1m[35mIntegerBinOp[39m[0m 
                              ([1m[35mVar[39m[0m 2 [33mn[39m) [1m[32m
                              Sub[39m[0m 
                              ([1m[35mCast[39m[0m ([1m[35mIntegerConstant[39m[0m [36m1[39m 
                                 ([1m[35mInteger[39m[0m 4 [])) [1m[32m
                                 IntegerToInteger[39m[0m 
                                 ([1m[35mInteger[39m[0m 8 []) ()) 
                              ([1m[35mInteger[39m[0m 8 []) ()))] 
                        ([1m[35mInteger[39m[0m 8 []) () ()) [1m[32m
                     Add[39m[0m 
                     ([1m[35mFunctionCall[39m[0m 1 [33mfib[39m () [
                        (
                           ([1m[35mIntegerBinOp[39m[0m 
                              ([1m[35mVar[39m[0m 2 [33mn[39m) [1m[32m
                              Sub[39m[0m 
                              ([1m[35mCast[39m[0m ([1m[35mIntegerConstant[39m[0m [36m2[39m 
                                 ([1m[35mInteger[39m[0m 4 [])) [1m[32m
                                 IntegerToInteger[39m[0m 
                                 ([1m[35mInteger[39m[0m 8 []) ()) 
                              ([1m[35mInteger[39m[0m 8 []) ()))] 
                        ([1m[35mInteger[39m[0m 8 []) () ()) 
                     ([1m[35mInteger[39m[0m 8 []) ()) ()) 
               ([1m[35mReturn[39m[0m)] 
               ([1m[35mVar[39m[0m 2 [33m_lpython_return_variable[39m) [1m[32m
               Source[39m[0m [1m[32m
               Public[39m[0m [1m[32m
               Implementation[39m[0m ()), 
         main: 
            ([1m[35mSubroutine[39m[0m 
               ([33mSymbolTable[39m 
                  3 
                  {
                     ans: 
                        ([1m[35mVariable[39m[0m 
                           3 
                           ans [1m[32m
                           Local[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mInteger[39m[0m 8 []) [1m[32m
                           Source[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .false.), 
                     x: 
                        ([1m[35mVariable[39m[0m 
                           3 
                           x [1m[32m
                           Local[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mInteger[39m[0m 8 []) [1m[32m
                           Source[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .false.)
                  }) 
               main [] [
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 3 [33mx[39m) 
                  ([1m[35mCast[39m[0m ([1m[35mIntegerConstant[39m[0m [36m40[39m 
                     ([1m[35mInteger[39m[0m 4 [])) [1m[32m
                     IntegerToInteger[39m[0m 
                     ([1m[35mInteger[39m[0m 8 []) ()) ()) 
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 3 [33mans[39m) 
                  ([1m[35mFunctionCall[39m[0m 1 [33mfib[39m () [
                     (
                        ([1m[35mVar[39m[0m 3 [33mx[39m))] 
                     ([1m[35mInteger[39m[0m 8 []) () ()) ()) 
               ([1m[35mPrint[39m[0m () [
                  ([1m[35mVar[39m[0m 3 [33mans[39m)] () ()) 
               ([1m[35mAssert[39m[0m 
                  ([1m[35mIntegerCompare[39m[0m 
                     ([1m[35mVar[39m[0m 3 [33mans[39m) [1m[32m
                     Eq[39m[0m 
                     ([1m[35mCast[39m[0m ([1m[35mIntegerConstant[39m[0m [36m102334155[39m 
                        ([1m[35mInteger[39m[0m 4 [])) [1m[32m
                        IntegerToInteger[39m[0m 
                        ([1m[35mInteger[39m[0m 8 []) ()) 
                     ([1m[35mLogical[39m[0m 4 []) ()) ())] [1m[32m
               Source[39m[0m [1m[32m
               Public[39m[0m [1m[32m
               Implementation[39m[0m () .false. .false.)
      }) 
   [
   ([1m[35mSubroutineCall[39m[0m 1 [33mmain[39m () [] ())])
(TranslationUnit (SymbolTable 1 {_lpython_main_program: (Subroutine (SymbolTable 5 {}) _lpython_main_program [] [(SubroutineCall 1 main () [] ())] Source Public Implementation () .false. .false.), fib: (Function (SymbolTable 2 {_lpython_return_variable: (Variable 2 _lpython_return_variable ReturnVar () () Default (Integer 8 []) Source Public Required .false.), n: (Variable 2 n In () () Default (Integer 8 []) Source Public Required .false.)}) fib [(Var 2 n)] [(If (IntegerCompare (Var 2 n) Lt (Cast (IntegerConstant 2 (Integer 4 [])) IntegerToInteger (Integer 8 []) ()) (Logical 4 []) ()) [(= (Var 2 _lpython_return_variable) (Var 2 n) ()) (Return)] []) (= (Var 2 _lpython_return_variable) (IntegerBinOp (FunctionCall 1 fib () [((IntegerBinOp (Var 2 n) Sub (Cast (IntegerConstant 1 (Integer 4 [])) IntegerToInteger (Integer 8 []) ()) (Integer 8 []) ()))] (Integer 8 []) () ()) Add (FunctionCall 1 fib () [((IntegerBinOp (Var 2 n) Sub (Cast (IntegerConstant 2 (Integer 4 [])) IntegerToInteger (Integer 8 []) ()) (Integer 8 []) ()))] (Integer 8 []) () ()) (Integer 8 []) ()) ()) (Return)] (Var 2 _lpython_return_variable) Source Public Implementation ()), main: (Subroutine (SymbolTable 3 {_lpython_return_variable_fib: (Variable 3 _lpython_return_variable_fib Local () () Default (Integer 8 []) Source Public Required .false.), ans: (Variable 3 ans Local () () Default (Integer 8 []) Source Public Required .false.), n_fib: (Variable 3 n_fib Local () () Default (Integer 8 []) Source Public Required .false.), x: (Variable 3 x Local () () Default (Integer 8 []) Source Public Required .false.), ~empty_block: (Block (SymbolTable 6 {}) ~empty_block [])}) main [] [(= (Var 3 x) (Cast (IntegerConstant 40 (Integer 4 [])) IntegerToInteger (Integer 8 []) ()) ()) (= (Var 3 n_fib) (Var 3 x) ()) (If (IntegerCompare (Var 3 n_fib) Lt (Cast (IntegerConstant 2 (Integer 4 [])) IntegerToInteger (Integer 8 []) ()) (Logical 4 []) ()) [(= (Var 3 _lpython_return_variable_fib) (Var 3 n_fib) ()) (GoTo 1)] []) (= (Var 3 _lpython_return_variable_fib) (IntegerBinOp (FunctionCall 1 fib () [((IntegerBinOp (Var 3 n_fib) Sub (Cast (IntegerConstant 1 (Integer 4 [])) IntegerToInteger (Integer 8 []) ()) (Integer 8 []) ()))] (Integer 8 []) () ()) Add (FunctionCall 1 fib () [((IntegerBinOp (Var 3 n_fib) Sub (Cast (IntegerConstant 2 (Integer 4 [])) IntegerToInteger (Integer 8 []) ()) (Integer 8 []) ()))] (Integer 8 []) () ()) (Integer 8 []) ()) ()) (GoTo 1) (BlockCall 1 3 ~empty_block) (= (Var 3 ans) (Var 3 _lpython_return_variable_fib) ()) (Print () [(Var 3 ans)] () ()) (Assert (IntegerCompare (Var 3 ans) Eq (Cast (IntegerConstant 102334155 (Integer 4 [])) IntegerToInteger (Integer 8 []) ()) (Logical 4 []) ()) ())] Source Public Implementation () .false. .false.), main_program: (Program (SymbolTable 4 {}) main_program [] [(SubroutineCall 1 _lpython_main_program () [] ())])}) [])
